lib/rubocop/cop/rspec/predicate_matcher.rb

```diff
@@ -150,11 +150,18 @@ def check_explicit(node) # rubocop:disable Metrics/MethodLength
 
           predicate_matcher?(node) do |actual, matcher|
             add_offense(node, message: message_explicit(matcher)) do |corrector|
+              next if uncorrectable_matcher?(node, matcher)
+
               corrector_explicit(corrector, node, actual, matcher, matcher)
             end
           end
         end
 
+        def uncorrectable_matcher?(node, matcher)
+          str_args = matcher.arguments.select(&:str_type?)
+          str_args.any?(&:heredoc?) && !same_line?(node, matcher)
```

Comment:
what if the heredoc contains variables? wouldn't it be dstr_type and still contain heredoc nodes? I suggest adding dstr and xstr here

Reply:
Thank you for catching this. I updated this PR.